pandas/core/generic.py

```diff
@@ -2060,7 +2064,13 @@ def sample(self, n=None, frac=None, replace=False, weights=None, random_state=No
             raise ValueError("A negative number of rows requested. Please provide positive value.")
 
         locs = rs.choice(axis_length, size=n, replace=replace, p=weights)
-        return self.take(locs, axis=axis)
+        result = self.take(locs, axis=axis)
+
+        # Make sure not returning a view. 
+        if not result._is_view:
+            result = result.copy()
```

Comment:
test for this too (use a frame with a single dtype, then one with a multi-dtype).